lib/wcc/contentful/model.rb

```diff
@@ -0,0 +1,24 @@
+
+# frozen_string_literal: true
+
+class WCC::Contentful::Model
+  extend WCC::Contentful::Helpers
+  extend WCC::Contentful::ModelValidators
+
+  class << self
+    attr_accessor :store
+  end
+
+  def self.all_models
+    WCC::Contentful::Model.constants(false).map { |k| WCC::Contentful::Model.const_get(k) }
+  end
+
+  def self.find(id, context = nil)
```

Comment:
could be defined in `inherited` block do prevent base class find

Reply:
I need to have a base class find because inside the dynamic model generation, when we resolve a link I don't actually know what content type the link is.  This encapsulates that logic.

Reply:
The key is this part:

```ruby
    content_type = content_type_from_raw(raw)
    const = WCC::Contentful::Model.const_get(constant_from_content_type(content_type))
    const.new(raw, context)
```

There's already a derived class `find(id)` which doesn't need to do that "const_get" stuff.  But I need to do it when I resolve links.

Reply:
https://zube.io/watermarkchurch/development/c/2235